MonoGame.Framework/Graphics/Effect/EffectParameterCollection.cs

```diff
@@ -7,10 +7,21 @@ public class EffectParameterCollection : IEnumerable<EffectParameter>
         internal static readonly EffectParameterCollection Empty = new EffectParameterCollection(new EffectParameter[0]);
 
         private readonly EffectParameter[] _parameters;
+        private readonly Dictionary<string, int> _lookupDictionary = new Dictionary<string, int>();
 
         internal EffectParameterCollection(EffectParameter[] parameters)
         {
             _parameters = parameters;
+            for (int i = 0; i < _parameters.Length; i++)

```

Comment:
Does `Dictionary` have a constructor that lets you reserve capacity?  This would be better than letting it grow as you add new keys.

Reply:
Yup. Right.
On it.